src/common/playlists/PlayMeta.jsx

```diff
@@ -1,78 +1,83 @@
-import { cloneElement, useEffect, useState, Suspense, lazy } from "react";
+import { useEffect, useState, Suspense } from "react";
 import { Helmet } from "react-helmet";
 import * as plays from "plays";
 import { useParams } from "react-router-dom";
-import underDevelopment from 'images/underdevelpoment.png'
 import { submit } from "common/services/request";
 import Loader from "common/spinner/spinner";
-import { toKebabCase, toPascalcase, toTitleCase } from "common/services/string";
-import { FetchPlaysByID } from "common/services/request/query/fetch-plays";
-import {PageNotFound} from 'common'
+import { toTitleCase } from "common/services/string";
+import { FetchPlaysByNameAndUser } from "common/services/request/query/fetch-plays";
+import { PageNotFound } from "common";
 
 function PlayMeta() {
-  const [loading, setLoading] = useState(true)
-  const [play, setPlay] = useState({})
-  const [isError, setIsError] = useState(false)
-  let { playid } = useParams(); // return the parameter of url
+  const [loading, setLoading] = useState(true);
+  const [play, setPlay] = useState({});
+  const [isError, setIsError] = useState(false);
+  let { playname, username } = useParams(); // return the parameter of url
+  const [metaImage, setMetaImage] = useState();
 
   useEffect(() => {
-    submit(FetchPlaysByID(playid)).then(res => {
-      setPlay(res[0])
-      setLoading(false)
-    }).catch(err => {
-      setIsError(true)
-      setLoading(false)
-    })
-  },[playid])
+    submit(FetchPlaysByNameAndUser(decodeURI(playname), decodeURI(username)))
+      .then((res) => {
+          const play_obj = res[0];
+          setPlay(play_obj);
+          setLoading(false);
+      })
+      .catch((err) => {
+        setIsError(true);
+        setLoading(false);
+      });
+  }, [playname, username]);
 
   if (loading) {
     return <Loader />;
-   }
-   if (isError) {
+  }
+  if (isError || !play) {
     return <PageNotFound />;
-   }
-  
-   const renderPlayComponent = () => {
-    const Comp = plays[play.component || toTitleCase(play.name).replace(/ /g,'')]
-    return <Comp {...play}/>
-   }
+  }
+
+  const renderPlayComponent = () => {
+    const Comp =
+      plays[play.component || toTitleCase(play.name).replace(/ /g, "")];
+    return <Comp {...play} />;
+  };
 
   return (
     <>
       <Helmet>
-        <meta name='description' content={play.description} />
-        <meta property='og:title' content={play.name} />
-        <meta property='og:description' content={play.description} />
-        {play.cover && (
+        <meta name="description" content={play.description} />
+        <meta property="og:title" content={play.name} />
+        <meta property="og:description" content={play.description} />
+        {metaImage && (
           <meta
-            property='og:image'
-            content={play.cover}
-            data-react-helmet='true'
+            property="og:image"
+            content={metaImage}
```

Comment:
If we use metaImage, it will also breast for the ones that are taking cover from DB right? Why don't we use the play.cover instead?

Reply:
Thanks for catching it. It was a residue from the previous implementation.